compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -355,6 +374,7 @@ public void visit(BLangPackage pkgNode, AnalyzerData data) {
             data.env = lambdaFunction.capturedClosureEnv;
             analyzeDef(lambdaFunction.function, data);
         }
+        validateClientDeclarations(pkgNode, data);
 
```

Comment:
Intentionally left this new line as is, since the preceding logic (including this addition) applies to analyzing a single module, whereas the next line calls analyze for the testable modules.

Reply:
+1